### PR TITLE
Feat/kn/add map layout component #33

### LINKED ISSUE #33: Add Map Layout Component
- [ ] When in this view we see a Header with some Text and a Button to go to the `chat` view. 
- [ ] Then in the center, we see a Map and Address divs.
- [ ] At the bottom we see a Footer ver with Text and a Button to redirect to WhatsApp:
- [ ] Data can be passed as props

image

### PR DESCRIPTION
**Pull request by Saikilito**
image

image

/test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "styled-components": "~5.3.1",
     "styled-normalize": "~8.0.7",
     "subscriptions-transport-ws": "~0.9.19",
+    "typescript": "^4.4.3",
```
Comment (Israel-Laguan) [inline package.json:15]: @ginnaaceror please confirm if you are OK with using typescript

File: src/pages/index.js [hunk 2]
```diff
@@ -25,6 +26,9 @@ const Pages = () => (
       <Route path="/confirm/:id?">
         <Confirm />
       </Route>
+      <Route path="/test">
+        <Test />
```
Comment (Israel-Laguan) [inline src/pages/index.js]: please remove this debugging artifact

File: src/pages/index.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ import Delivery from './Delivery'
 import Error from './Error'
 import Home from './Home'
 import NotFound from './NotFound'
+import Test from './TEST'
```
Comment (Israel-Laguan) [inline src/pages/index.js]: please remove this debugging artifact

File: package.json [hunk 4]
```diff
@@ -74,7 +79,8 @@
     "stylelint": "13.13.1",
     "stylelint-config-standard": "22.0.0",
     "stylelint-processor-styled-components": "1.10.0",
-    "stylelint-webpack-plugin": "2.2.2"
+    "stylelint-webpack-plugin": "2.2.2",
+    "tslint": "^6.1.3"
```
Comment (Israel-Laguan) [inline package.json]: `tslint` is deprecated, please not use it
image

reference: https://www.infoq.com/news/2019/02/tslint-deprecated-eslint/#:~:text=Palantir%2C%20the%20creators%20of%20TSLint,linting%20solution%20for%20TypeScript%20users.

**Review by Israel-Laguan — CHANGES_REQUESTED** [review]
Also, why did you erase these files? 
image

These are needed in the remaining file so its breaking the app.

Also you didn't include images in `src/asset` back or forgot to add to commit, please check
image

**Comment by Israel-Laguan** [comment]
I don't see css code for Layout, please check or commit unstaged changes @Saikilito

File: package.json (hunk 1 above)
Comment (ginnaaceror) [reply package.json:15]: I prefer ES6.

**Comment by Saikilito** [comment]
@Israel-Laguan  make sure you have the folder with the full name, the items were not deleted just renamed.

File: src/pages/index.js (hunk 3 above)
Comment (Saikilito) [reply src/pages/index.js]: removed

File: src/layouts/map.layout.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { ReactNode } from 'react';
+
+import { MapUpperBar, MapLowerBar } from './components';
+
+type IChildrenType = {
+  children: ReactNode;
+};
+
+// TODO: the spelling gives problems with spanish
```
Comment (Israel-Laguan) [inline src/layouts/map.layout.tsx]: Just add Spanish to the available languages by using an extension

https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker-spanish

File: .prettierrc.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // .prettierrc.js
 module.exports = {
   printWidth: 120,
-  semi: false,
+  semi: true,
```
Comment (Israel-Laguan) [inline .prettierrc.js:4]: why?

File: cspell.config.json [hunk 7]
```diff
@@ -1,24 +1,15 @@
...
+  "flagWords": ["hte"],
   "ignorePaths": [
     "node_modules/**",
     "public/**",
     "src/service-worker.js",
     "src/serviceWorkerRegistration.js",
     "src/reportWebVitals.js",
-    "package.json"
+    "package.json",
+    ".git/**"
```
Comment (Israel-Laguan) [inline cspell.config.json]: Good finding, what problems this was doing?

File: package.json [hunk 8]
```diff
@@ -30,8 +31,8 @@
     "start": "react-app-rewired start",
     "build": "react-app-rewired build",
     "eject": "react-app-rewired eject",
-    "spelling": "cspell run src/**/*.js",
-    "lint:css": "stylelint --config .stylelintrc.json './**/*.styled.jsx'; echo 'CSS fine! 💅'",
+    "spelling": "cspell run src/**/*.tsx",
+    "lint:css": "stylelint --config .stylelintrc.json './**/*.styled.tsx'; echo 'CSS fine! 💅'",
```
Comment (Israel-Laguan) [inline package.json]: just add the extension by using regex please, like this:

`src/**/*.{js,jsx,ts,tsx}`

File: src/layouts/components/MapUpperBar/MapUpperBar.style.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import styled from 'styled-components';
+import { colors, breakpoints } from '../../../styles';
+
+export const MapUpperBarStyle = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  background-color: ${colors.default.blue08};
```
Comment (Israel-Laguan) [inline src/layouts/components/MapUpperBar/MapUpperBar.style.tsx]: what about `var(--primary);`? Please try it

File: .prettierrc.js (hunk 6 above)
Comment (Saikilito) [reply .prettierrc.js:4]: the types and interface for typescript require ";" the final of the line

Comment (Israel-Laguan) [reply .prettierrc.js:4]: Ok, got it, thanks for explain

File: cspell.config.json (hunk 7 above)
Comment (Saikilito) [reply cspell.config.json]: it backs

**Comment by Saikilito** [comment]
This branch iss squashed

File: src/layouts/components/MapLowerBar/MapLowerBar.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        <CenterContainer>
+          <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', padding: '1rem' }}>
+            <div>
+              <img src={GpsIcon} alt="" />
+            </div>
+            <div style={{ height: '100%' }}>
+              <Paragraph color={colors.colors.yellow05}>Av. Apurímac 1020, Santiago de chile</Paragraph>
+            </div>
+          </div>
+          <div style={{ display: 'flex', width: '100%', justifyContent: 'space-evenly', alignItems: 'center' }}>
```
Comment (Israel-Laguan) [inline src/layouts/components/MapLowerBar/MapLowerBar.tsx]: This one have too much inline styles, please make it a Component using styled components

File: src/layouts/components/MapLowerBar/MapLowerBar.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+import AssistantIcon from '../../../assets/assistant.png';
+import GpsIcon from '../../../assets/gps-icon.png';
+
+export const MapLowerBar = () => {
+  return (
+    <>
+      <MapLowerBarStyle>
+        <CenterContainer>
+          <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', padding: '1rem' }}>
```
Comment (Israel-Laguan) [inline src/layouts/components/MapLowerBar/MapLowerBar.tsx]: This one have too much inline styles, please make it a Component using styled components

File: src/layouts/components/MapUpperBar/MapUpperBar.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import { IMapUpperBarTypeInput } from './MapUpperBar.type';
+import { Paragraph, ParagraphWithoutMobile, TitleLG, ImageControl } from '../../../components';
+
+import ArrowLeft from '../../../assets/arrow-left.svg';
+import ChatIcon from '../../../assets/chat.png';
+
+export const MapUpperBar = ({ text, subtext }: IMapUpperBarTypeInput) => {
+  return (
+    <MapUpperBarStyle>
+      <div style={{ display: 'flex', alignItems: 'center', color: 'white' }}>
```
Comment (Israel-Laguan) [inline src/layouts/components/MapUpperBar/MapUpperBar.tsx]: This one have too much inline styles, please make it a Component using styled components

File: src/styles/text.ts [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+  `,
+  black: css`
+    color: var(--dasher-black);
+  `,
+  white: css`
+    color: var(--dasher-white);
+  `,
+  danger: css`
+    color: var(--dasher-danger);
+  `,
```
Comment (Israel-Laguan) [inline src/styles/text.ts]: Please be aware these were fixed in Ginna's PR, [HERE](https://github.com/Azordev/dasher-user/blob/e41aa0b119f4dbf679e971a389a88c14e56f7e83/src/styles/text.js).

Change it to be like these

File: src/styles/text.ts [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+  test: 'color: red;',
```
Comment (Israel-Laguan) [inline src/styles/text.ts]: Remove this line please

**Review by Israel-Laguan — CHANGES_REQUESTED** [review]

File: src/styles/text.ts (hunk 13 above)
Comment (Saikilito) [reply src/styles/text.ts]: this change is only prettier

Comment (Israel-Laguan) [reply src/styles/text.ts]: No, look that name of variable doesn't includes 'dasher'. I am asking you to update your code according to the correct names, can you do it?

File: src/styles/text.ts (hunk 14 above)
Comment (Saikilito) [reply src/styles/text.ts]: removed

File: src/components/index.tsx [hunk 15]
```diff
@@ -0,0 +1,4 @@
+export * from './Marker';
+export * from './Texts';
+export * from './Container';
```
Comment (Israel-Laguan) [inline src/components/index.tsx:3]: I checked and Container is not in components, can you check, please?
image

Comment (Israel-Laguan) [reply src/components/index.tsx:3]: This is breaking the app

Comment (Saikilito) [reply src/components/index.tsx:3]: resolved

File: src/pages/Delivery.jsx [hunk 16]
```diff
@@ -1,44 +1,53 @@
...
+  const { id } = useParams();
+  const history = useHistory();
 
   if (!id) {
-    history.push('/check')
+    history.push('/check');
   }
 
   return (
     <div>
```
Comment (Israel-Laguan) [inline src/pages/Delivery.jsx]: Please include here the layout you just created so we can test @Saikilito

Comment (Israel-Laguan) [reply src/pages/Delivery.jsx]: Currently looks like this
image

Comment (Saikilito) [reply src/pages/Delivery.jsx]: included

File: README.md [hunk 17]
```diff
@@ -148,6 +148,10 @@ To learn React, check out the [React documentation](https://reactjs.org/).
 |:---:|:---:|
 | ![author-pic][] | ![banner][] |
 
+| [Kembert Nieves][author-github] | ![email-icon][] Email me to [contact@kembertnieves.com][author-email] / ![linkedin-icon][] Connect to [my Linkedin][author-linkedin] |
```
Comment (Israel-Laguan) [inline README.md:151]: Linkedin link is showing my profile, please update

File: README.md [hunk 18]
```diff
@@ -148,6 +148,10 @@ To learn React, check out the [React documentation](https://reactjs.org/).
...
+|:---:|:---:|
+| ![author-pic][] | ![banner][] |
```
Comment (Israel-Laguan) [inline README.md:153]: Photo is mine, please update. Also banner have my name

File: package.json (hunk 1 above)
Comment (Israel-Laguan) [reply package.json:15]: @ginnaaceror can you try on your pc please?

File: src/layouts/components/MapLowerBar/MapLowerBar.styled.tsx [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+  position: relative;
+  bottom: 0;
+
+  transform: translateX(50%);
+
+  border-radius: 30px 30px 0px 0px;
+
+  @media only screen and (max-width: ${breakpoints.breakpoints.md}) {
+    height: 35%;
+    width: 105%;
```
Comment (Israel-Laguan) [inline src/layouts/components/MapLowerBar/MapLowerBar.styled.tsx:23]: why is bigger than 100%?

File: package.json (hunk 1 above)
Comment (ginnaaceror) [reply package.json:15]: Hi @Israel-Laguan I can see it well on my pc.

Comment (ginnaaceror) [reply package.json:15]: finally we are going to use TS or ES6?

File: src/components/Container/EvenlyContainer.tsx [hunk 20]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
+
+export const EvenlyContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-evenly;
+
```
Comment (ginnaaceror) [inline src/components/Container/EvenlyContainer.tsx:7]: hi @Saikilito on line 7 why are you leaving this space?  ***it's curiosity :)***

**Review by ginnaaceror — APPROVED** [review]

**Pull request closed without merging**